src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -1051,6 +1051,11 @@ private static void DeleteTargetAndTempPaths(string targetPath, string targetTem
                     repositorySignatureInfo,
                     clientPolicyContext.VerifierSettings);
 
+                if (signedPackageReader.CanVerifySignedPackages(verificationSettings))
```

Comment:
Shouldn't this condition be flipped? 

Can you add a test for this specific scenario?

Reply:
this PR already adds tests for this scenarios https://github.com/NuGet/NuGet.Client/pull/2471/files#diff-422327819c7fa777b2ea7e9e05187dc4R1893